bot/api.py

```diff
@@ -52,33 +52,49 @@ def __init__(self, loop: asyncio.AbstractEventLoop, **kwargs):
 
         self._ready = asyncio.Event(loop=loop)
         self._creation_task = None
-        self._session_args = kwargs
+        self._default_session_kwargs = kwargs
 
         self.recreate()
 
     @staticmethod
     def _url_for(endpoint: str) -> str:
         return f"{URLs.site_schema}{URLs.site_api}/{quote_url(endpoint)}"
 
-    async def _create_session(self) -> None:
-        """Create the aiohttp session and set the ready event."""
-        self.session = aiohttp.ClientSession(**self._session_args)
+    async def _create_session(self, **session_kwargs) -> None:
+        """
+        Create the aiohttp session with `session_kwargs` and set the ready event.
+
+        `session_kwargs` is merged with `_default_session_kwargs` and overwrites its values.
+        If an open session already exists, it will first be closed.
+        """
+        await self.close()
+        self.session = aiohttp.ClientSession(**{**self._default_session_kwargs, **session_kwargs})
```

Comment:
Do we ever need to save the default values after they have been overwritten? then dict-merging might not be the best option. ChainMap would fix this issue as long as you have an empty dict as the first dict in the chain.

Reply:
> Do we ever need to save the default values after they have been overwritten?

No. The merged dictionary being created is one useful once since the `session_kwargs` as passed as an argument and may be different each call.

Reply:
Thanks for the clarification.